### ISSUE TITLE
didReceiveIncomingPushWithPayload called even after unregistering for VoIP notifications

### ISSUE THREAD

**Issue by zubairuddin** [twilio/voice-quickstart-swift#51]
Hi Twilio Support,

We are using Twilio ProgrammableVoice SDK version 2.0.0 - beta9 in our app. We only allow outgoing and incoming calls when user is logged-in on the app. I register for the VoIP notifications after login and unregister on logout. But even after logging out, I'm getting a call to `didReceiveIncomingPush` and `callInviteReceived` methods.

Can you please let me know what could be the reason for this.

Here is the code I'm using for unregistering 

```
if let deviceToken = ApplicationManager.shared.strDeviceTokenForVoIPCalls, let accessToken = ApplicationManager.shared.twilioToken {
                        TwilioVoice.sharedInstance().unregister(withAccessToken: accessToken, deviceToken: deviceToken) { (error) in
                            if (error != nil) {
                                NSLog("An error occurred while unregistering: \(String(describing: error?.localizedDescription))")
                            }
                            else {
                                ApplicationManager.shared.isTwilioConfigured = false
                                NSLog("Successfully unregistered from VoIP push notifications.")
                            }
                        }
                    }

```
I'm even getting console log saying `Successfully unregistered from VoIP push notifications`

But once I'm on the login screen, I'm still getting incoming call notifications.

Here are more details
```
iOS version - 10.3.2
Device - iPhone 6 Plus
ProgrammableVoice SDK version - 2.0.0 - beta9
```
Thanks

**Comment by bchen-twilio** [comment]
Hi @zubairuddin 

Thanks for reporting the issue and the detailed descriptions 👍 

Sorry that you are experiencing the unexpected incoming push notification issue. Could you help me with a couple of things?
- Could you try the same unregistration steps again and then use another Twilio Voice client to call this unregistered identity. I can run the Twilio Call-SID (`CA{32}`, which can be found in the caller's log) and dig up more details how the call went.
- This is based my experience while developing this SDK: (I am assuming that this issue happened on one or some specific devices) is it possible that multiple VoIP Service certificates or Twilio Push Credentials were used on these devices for registration? the server DB might still have the old bindings with **the same identity** but different push device token associated with any VoIP cert previously used. Twilio Voice service will attempt to send notifications to all the bindings with the client identity being called.

I believe I still have your Twilio account SID in my email inbox, but just want to confirm, is the account SID ending with `38be` the current account you are using? I will reach out to the Twilio Notify team and see if they can purge old registered bindings under your account.

Thanks.
-bobie

**Comment by zubairuddin** [comment]
Hi @bchen-twilio 

Thanks for your response.

- I did try the same process again on different clients, but the results were same. I got incoming call notifications even after unregistering.

- Yes it's possible, for testing our build we are using Push credentials SID created on our demo Twilio account, and once we are ready to provide build to our client, we change the Push credentials SID on our server with those created with Paid Twilio account.

For testing, I'm still using the same Account SID, the one ending with `38be`.

Thanks,
Zubair

**Comment by bchen-twilio** [comment]
@zubairuddin Thanks for clarifying my questions.

The design policy of Twilio should eliminate the case that push notifications being sent to different Twilio account (SID).

To make sure unregistering works, could you try these steps?
- launch the app and register with an identity that has never been used on your testing device before
- then unregister that identity from the notification service
- try to call this new identity and see if you are still getting incoming call push notifications.

Thanks
-bobie

**Comment by zubairuddin** [comment]
Hi @bchen-twilio 

Replying late since I was busy on working on some other modules in my app.

I did try the steps you suggested:

-Registered with Twilio with an identity that was never used before
-Unregistered the identity by logging out of the app.
-Tried to call this identity
 
However, I was still getting the incoming call even after unregistering.

Please let me know what could be the issue and the steps to resolve it.

Thanks.

**Comment by bchen-twilio** [comment]
Hi @zubairuddin 

How are things? We've just released the 2.0.0 Voice SDK before the holidays and I am wondering if you have the chance to take a look at the latest APIs and sample code?

I am going to close this ticket for now assuming the original issue is resolved but please feel free to reopen or create a new one if you run into any problem in the future.

Thanks.
-bobie
